src/Controller/TranscriptionsController.php

```diff
@@ -84,18 +84,13 @@ public function reset($sentenceId, $script) {
             );
         }
 
-        if (!$saved) {
-            return $this->response->withStatus(400, __('The transcription could not be reset.'));
-        }
-
         $acceptsJson = $this->request->accepts('application/json');
         if ($acceptsJson) {
-            $transcription = $this->Transcriptions->findTranscription($sentenceId, $script);
-            $this->set('result', $transcription);
-            $this->loadComponent('RequestHandler');
-            $this->set('_serialize', ['result']);
-            $this->RequestHandler->renderAs($this, 'json');
+            return $this->returnTranscriptionData($saved, $sentenceId, $script);
         } else {
+            if (!$saved) {
+                return $this->response->withStatus(400, 'Bad transcription');
+            }
             $this->setViewVars(array_filter(array($saved)), $sentenceId, $sentence);
             $this->render('view');
```

Comment:
Isn’t this going to prevent `$this->setViewVars()` and` $this->render()` from being executed if the transcription cannot be reset?

Reply:
Yes, but it's how things are done in the current code:
https://github.com/Tatoeba/tatoeba2/blob/dev/src/Controller/TranscriptionsController.php#L87

I'm not entirely sure of the impacts of moving the `if` block at the end so I kept it like that.

Reply:
I see. Never mind my comment.